opennms-container/minion/assets/entrypoint.sh

```diff
@@ -149,12 +152,42 @@ applyOverlayConfig() {
   fi
 }
 
+applyConfd() {
+  if [ -f "${CONFD_KEY_STORE}" ]; then
+    echo "Found a configuration key store, applying configuration via confd."
+    runConfd
+  else
+    echo "No configuration key store present, skipping confd configuration."
+  fi
+}
+
 start() {
     export KARAF_EXEC="exec"
     cd ${MINION_HOME}/bin
     exec ./karaf server
 }
 
+runConfd() {
+  # Create any directories that confd might write to
+  while IFS= read -r dir; do
+    local dirToCreate="$MINION_HOME"/"$dir"
+    echo "Creating $dirToCreate so confd can write to it"
+    mkdir -p "$dirToCreate"
+  done < "$CONFD_CONFIG_DIR"/directories
+
+  "$CONFD_BIN" -log-level debug -onetime -backend=file -file "$CONFD_KEY_STORE" -confdir "$CONFD_CONFIG_DIR"
```

Comment:
Just to be sure, we nail it down to use only the minion-config.yaml file and don't allow other configuration backends right? Just as a quick idea, we could gain some flexibility using a `confd.toml` and call the binary with `-config-file` 

```
confdir = "/etc/confd"
backend = "file"
file = "/opt/minion/minion-config.yaml"
```
I don't worry much - no blocker for and really just a minor comment.

Reply:
I didn't mean to lock it down to a file. Agreed using a confd.toml will allow us to be more flexible since this file could be optionally mounted to changed the config. Will use it.